server/services/caService.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @typedef {import("../../types/delius").DeliusClient} DeliusClient
+ * @typedef {import("../../types/licences").CaService} CaService
+ */
+
+/**
+ * @param {DeliusClient} deliusClient
+ * @returns {CaService} caService
+ */
+
+const logger = require('../../log.js')
+const roService = require('./roService')
+
+module.exports = function createCaService(deliusClient, nomisClientBuilder) {
+  async function getLduForRo(bookingId, token) {
+    const ro = await roService(deliusClient, nomisClientBuilder).findResponsibleOfficer(bookingId, token)
```

Comment:
Services should only be created once and live the entire lifecycle of the application - not created per request. 
You can pass the created service in to the `createCaService`

Reply:
I used the example of roService in creating the service. 
...

The caService was created in server/index.js and passed to createApp

So do you mean I shouldn't wrap the getLduForRo in createCaService, rather just export it on it's own, like this...
module.exports = async function getLduForRo(bookingId, token) {
const { lduCode, message } = 
await roService(deliusClient,nomisClientBuilder)
.findResponsibleOfficer( bookingId, token  )
 etc, etc


?

Reply:
this comment is more about only passing `roService` rather than creating it in the method: 

`function createCaService(roService) {`

Reply:
oh, I see

Reply:
doh!